modules/core/04-channel/keeper/keeper_test.go

```diff
@@ -81,6 +83,97 @@ func (suite *KeeperTestSuite) TestGetAppVersion() {
+func (suite *KeeperTestSuite) TestGetAllChannelsWithPortPrefix() {
+
+	const (
+		secondChannelID        = "channel-1"
+		differentChannelPortID = "different-portid"
+	)
+
+	tests := []struct {
+		name             string
+		prefix           string
+		allChannels      []types.IdentifiedChannel
+		expectedChannels []types.IdentifiedChannel
+	}{
+		{
+			name:   "transfer channel is retrieved with prefix",
+			prefix: "tra",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.PortID, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel(differentChannelPortID, secondChannelID, types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{})},
+		},
+		{
+			name:   "matches port with full name as prefix",
+			prefix: transfertypes.TypeMsgTransfer,
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.PortID, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel(differentChannelPortID, secondChannelID, types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{})},
+		},
+		{
+			name:   "no ports match prefix",
+			prefix: "wont-match-anything",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.PortID, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel(differentChannelPortID, secondChannelID, types.Channel{}),
+			},
+			expectedChannels: nil,
+		},
+		{
+			name:   "empty prefix matches everything",
+			prefix: "",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.PortID, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel(differentChannelPortID, secondChannelID, types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.PortID, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel(differentChannelPortID, secondChannelID, types.Channel{}),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			for _, ch := range tc.allChannels {
+				suite.chainA.GetSimApp().GetIBCKeeper().ChannelKeeper.SetChannel(suite.chainA.GetContext(), ch.PortId, ch.ChannelId, types.Channel{})
+			}
+
+			ctxA := suite.chainA.GetContext()
+
+			actualChannels := suite.chainA.GetSimApp().GetIBCKeeper().ChannelKeeper.GetAllChannelsWithPortPrefix(ctxA, tc.prefix)
+
+			suite.Require().True(containsAll(tc.expectedChannels, actualChannels))
+		})
+	}
+}
+
+// containsAll verifies if all elements in the expected slice exist in the actual slice
+// independent of order.
+func containsAll(expected, actual []types.IdentifiedChannel) bool {
```

Comment:
I added this function as I saw that the order of the values returned in `GetAllChannels` is not necessarily in the order they were added.